engine/src/main/java/org/hibernate/search/engine/impl/AnalyzerReferenceRegistry.java

```diff
@@ -82,11 +80,59 @@ public T getAnalyzerReference(Class<?> analyzerClazz) {
 	}
 
 	public void initialize(Map<String, AnalyzerDef> analyzerDefinitions) {
-		strategy.initializeNamedAnalyzerReferences( referencesByName, analyzerDefinitions );
+		List<AnalyzerReference> references = new ArrayList<>();
```

Comment:
As far as I can tell, the actual problem is we create new instances of analyzers when dynamically adding new entities, even for analyzers we already discovered. So we end up having multiple references to the same analyzer, pointing to different instances of this analyzer. It's similar for Elasticsearch.
I guess it's not critical, unless you have dozens of analyzers and you dynamically add new entities every two seconds.

This is a pre-existing bug, though, not really the point of this PR. Solving it would require to initialize the `ConfigContext` analyzer registries with the content of the previous version of the search integrator.
Actually I did solve this bug in another branch I'm working on (for other reasons), but that branch is currently blocked because of [an unrelated, ongoing debate over what to do next](http://lists.jboss.org/pipermail/hibernate-dev/2017-January/015704.html).
You can see the solution on [my HSEARCH-2534 branch](https://github.com/yrodiere/hibernate-search/commits/HSEARCH-2534), the commit https://github.com/yrodiere/hibernate-search/commit/af1642f06de212ed4ab130cff7dafd2786cf49c6 in particular, and [here](https://github.com/yrodiere/hibernate-search/commit/af1642f06de212ed4ab130cff7dafd2786cf49c6#diff-a85290d5c89683cd208360a89c065a51R122) to be very precise.